url294e6c45f3_Ecdosis_NMergeNew_tgz-pJ8-TestJ8/src/edu/luc/nmerge/mvd/MVDError.java

```diff
@@ -0,0 +1,25 @@
+package edu.luc.nmerge.mvd;
+
+import java.io.FileWriter;
+import java.io.File;
+/**
+ * Log errors to a /tmp file
+ * OK, this is quick and dirty
+ */
+public class MVDError 
+{
+	public static void log( String message )
+	{
+		try
+		{
+			File tmpDir = new File( System.getProperty("java.io.tmpdir"));
+			File tmpFile = new File( tmpDir,"mvderror.log" );
+			FileWriter fw = new FileWriter(tmpFile,true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced at line 18. The resource is closed at line 19. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *fw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).